src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -838,11 +838,17 @@ public async Task DeleteBlockAsync(string blockHash)
       {
          SyncBlockInfo block = BlockByHash(blockHash);
 
-         FilterDefinition<InputTable> addrForInputFilter = Builders<InputTable>.Filter.Eq(addr => addr.BlockIndex, block.BlockIndex);
-         Task<DeleteResult> input = InputTable.DeleteManyAsync(addrForInputFilter);
+         FilterDefinition<InputTable> inputFilter = Builders<InputTable>.Filter.Eq(addr => addr.BlockIndex, block.BlockIndex);
+         var inputs = InputTable.FindSync(inputFilter).ToList();
 
-         FilterDefinition<OutputTable> addrForOutputFilter = Builders<OutputTable>.Filter.Eq(addr => addr.BlockIndex, block.BlockIndex);
-         Task<DeleteResult> output = OutputTable.DeleteManyAsync(addrForOutputFilter);
+         Task utxos = UtxoTable.InsertManyAsync(inputs.Select(_ => new UtxoTable
+            {
+               Address = _.Address, Outpoint = _.Outpoint, Value = _.Value, BLockIndex = _.BlockIndex
+            }))
+            .ContinueWith(task => InputTable.DeleteManyAsync(inputFilter));
```

Comment:
shouldn't this be async/await?

Reply:
ho its returning the task right?

Reply:
Yes, without the brackets and one line like this, is equal to { return InputTable... } ... if I'm not reviewing this too early.

Reply:
Sorry, this code is not ment to go to master, I'm going to change the logic for the entire delete because we don't have indexes on the tables we delete from